skins/laika/src/view_models/AddressData.ts

```diff
@@ -0,0 +1,39 @@
+export interface PrivateDonorName {
+    salutation: string,
+    firstName: string,
+    lastName: string
+}
+
+export interface CompanyDonorName {
+    company: string
+}
+
+export interface DonorAddress {
+    streetAddress: string,
+    postalCode: string,
+    city: string,
+}
+
+export interface DonorEmail {
+    email: string
+}
+
+export interface PrivateDonor {
+    name: PrivateDonorName,
+    address: DonorAddress,
+    email: DonorEmail
+}
+
+export interface CompanyDonor {
+    name: CompanyDonorName,
+    address: DonorAddress,
+    email: DonorEmail
+}
+
+export interface EmailDonor {
+    email: DonorEmail
+}
```

Comment:
`DonorEmail` is not needed. `email: string` would be fine.
Or alternatively drop all the email related interfaces as the email is always a string no matter what kind of donor it is. Then in `DonationConfirmationModel` it would be `email: string|null`

Reply:
Yeah, was also wondering if that might be better but I agree.